routers/api/v1/api.go

```diff
@@ -270,6 +270,15 @@ func reqRepoWriter(unitTypes ...unit.Type) func(ctx *context.APIContext) {
 	}
 }
 
+// reqRepoBranchWriter user should have a permission to write to a branch, or be a site admin
+func reqRepoBranchWriter(ctx *context.APIContext) {
+	options := web.GetForm(ctx).(api.FileOptionInterface)
```

Comment:
```suggestion
	options, ok := web.GetForm(ctx).(api.FileOptionInterface)
```
handle the case where it does not match the interface